__main__.py

```diff
@@ -1,42 +1,83 @@
-from simulator.modules import Clients, LoadBalancer, ServerWithGCI
-from log import get_logger, log_server_data, log_latency, log_percentiles
+from simulator.models import Clients, LoadBalancer, ServerWithGCI, Server
+from log import iniciate_csv_files, csv_writer
 from config import get_config
 import simpy, os, time, math
 
-def create_directory():
-    if not os.path.isdir("logs"):
-        os.mkdir("logs")
+def create_directory(logs_path, results_path):
+    if not os.path.isdir(logs_path):
+        os.mkdir(logs_path)
 
-def percentile(P, N):
-    return math.ceil((P / 100) * N)
+    if not os.path.isdir(results_path):
+        os.mkdir(results_path)
+
+def percentile(l, p):
+    """Returns the value within the list l representing pth percentile."""
+    s = sorted(l, key=lambda x: x._latency_time)
+    pos = math.ceil((p / 100) * len(s))
+    return s[pos - 1]._latency_time
+
+def log_latency(results_path, requests, scenario, load):
+    first_request = requests[0]._latency_time
+    last_request = requests[-1]._latency_time
+    average_latency = sum(request._latency_time for request in requests) / len(requests)
+
+    ordered_requests = sorted(requests, key=lambda x: x._latency_time)
```

Comment:
No need to order here. They are ordered inside the percentile function.

Reply:
My mistake. I forgot to remove. In fact, I'll remove from the percentile function because there it would order the list multiple times. Thanks again!